ohhamma/react-todo/src/App.js

```diff
@@ -0,0 +1,54 @@
+import { useState, useEffect } from 'react';
+import TodoContainer from './pages/Todo/TodoContainer';
+import TodoHead from './pages/Todo/TodoHead';
+import TodoForm from './pages/Todo/TodoForm';
+import TodoList from './pages/Todo/TodoList';
+import './App.css';
+import './pages/Todo/Todo.css';
+
+function App() {
+  const TODOS_KEY = "todos";
+
+  const saveTodos = () => {
+    localStorage.setItem(TODOS_KEY, JSON.stringify(todos));
+  }
+
+  const loadTodos = () => {
+    const savedTodos = localStorage.getItem(TODOS_KEY);
+    if (savedTodos !== null) {
+      const parsedTodos = JSON.parse(savedTodos);
+      return parsedTodos;
+    }
+    else return [];
+  }
+
+  const [todos, setTodos] = useState(loadTodos());
+  
+  const onAdd = (text) => {
+    setTodos([
+      ...todos,
+      {
+        id: Date.now(),
+        text
+      }
+    ])
+  }
+
+  const onDel = (id) => {
+    setTodos(todos.filter(todo => todo.id !== id));
+  }
+
+  useEffect(() => {
+    saveTodos();
+  })
```

Comment:
```suggestion
  useEffect(() => {
    saveTodos();
  }, [todos])
```

useEffects 훅을 쓸 때는 의존성 배열을 함께 넣어줄 수 있습니다. todos 상태가 변경되었을때 함수가 실행되도록 할 수 있어요. 의존성 배열이 없이 쓸때는 컴포넌트가 랜더링될때마다 매번 실행됩니다!!